src/lib/mina_base/snapp_generators.ml

```diff
@@ -284,13 +284,73 @@ let gen_balance_change ?balances_tbl (account : Account.t) =
 let gen_use_full_commitment : bool Base_quickcheck.Generator.t =
   Bool.quickcheck_generator
 
+let closed_interval_exact value =
+  Snapp_predicate.Closed_interval.{ lower = value; upper = value }
+
+let epoch_data_predicate
+    (epoch_data :
+      ( ( Frozen_ledger_hash.Stable.V1.t
+        , Currency.Amount.Stable.V1.t )
+        Epoch_ledger.Poly.Stable.V1.t
+      , Epoch_seed.Stable.V1.t
+      , State_hash.Stable.V1.t
+      , State_hash.Stable.V1.t
+      , Mina_numbers.Length.Stable.V1.t )
+      Snapp_predicate.Protocol_state.Epoch_data.Poly.t) :
+    Snapp_predicate.Protocol_state.Epoch_data.t =
+  let ledger =
+    let total_currency = epoch_data.ledger.total_currency in
+    Epoch_ledger.Poly.
+      { hash = Snapp_basic.Or_ignore.Check epoch_data.ledger.hash
```

Comment:
Could you use `Or_ignore.gen` everywhere? there could be `Ignore`s as well

Reply:
Done!